src/types.rs

```diff
@@ -218,3 +218,40 @@ impl<'prev, 'now> SubmitArgs<'prev, 'now> {
         }
     }
 }
+
+#[cfg(feature = "unstable")]
+#[repr(transparent)]
+pub struct BufRingEntry(sys::io_uring_buf);
+
+/// An entry in a buf_ring that allows setting the address, length and buffer id.
+#[cfg(feature = "unstable")]
+impl BufRingEntry {
+    /// Sets the entry addr.
+    pub fn set_addr(&mut self, addr: u64) {
+        self.0.addr = addr;
+    }
+    /// Returns the entry addr.
+    pub fn _addr(&mut self) -> u64 {
+        self.0.addr
+    }
+    /// Sets the entry len.
+    pub fn set_len(&mut self, len: u32) {
+        self.0.len = len;
+    }
+    /// Returns the entry len.
+    #[allow(clippy::len_without_is_empty)]
+    pub fn len(&mut self) -> u32 {
+        self.0.len
+    }
+    /// Sets the entry bid.
+    pub fn set_bid(&mut self, bid: u16) {
+        self.0.bid = bid;
+    }
+    /// Returns the entry bid.
+    pub fn bid(&mut self) -> u16 {
+        self.0.bid
+    }
+}
+
+/// The offset to the resv field of the io_uring_buf struct.
+pub const RING_BUFFER_TAIL_OFFSET: usize = 14;
```

Comment:
Sorry, I thought about it, I'm a little unhappy with this `TAIL_OFFSET`.

I consider an api like this

```rust
impl BufRingEntry {
    pub fn tail(ptr: *const BufRingEntry) -> *const u16 {
        unsafe {
            std::ptr::addr_of!((*ptr).resv)
        }
    }
}

Reply:
So it returns a pointer? Then I use that in the atomic update?

Reply:
Yes, you can cast to `AtomicU16`, or just return `*const AtomicU16` here

Reply:
I get this clippy warning
 ```
error: this public function might dereference a raw pointer but is not marked `unsafe`
   --> src/types.rs:259:34
    |
259 |             std::ptr::addr_of!((*ptr).0.resv)
    |                                  ^^^
    |
    = note: `#[deny(clippy::not_unsafe_ptr_arg_deref)]` on by default
```
Want me to make the whole function unsafe or allow the use to quiet clippy?

Reply:
I'm going with the former.

Reply:
Yes, please mark it as unsafe.